src/HotChocolate/Filters/src/Types.Filters/FilterInputTypeDescriptor.cs

```diff
@@ -137,95 +138,23 @@ public IFilterInputTypeDescriptor<T> Directive<TDirective>()
         {
+            definition = null;
             Type type = property.PropertyType;
 
             if (type.IsGenericType
-                && System.Nullable.GetUnderlyingType(type) is { } nullableType)
+                && System.Nullable.GetUnderlyingType(type) is Type nullableType)
             {
                 type = nullableType;
             }
 
-            if (type == typeof(string))
-            {
-                var field = new StringFilterFieldDescriptor(Context, property);
-                definition = field.CreateDefinition();
-                return true;
-            }
-
-            if (type == typeof(bool))
-            {
-                var field = new BooleanFilterFieldDescriptor(
-                    Context, property);
-                definition = field.CreateDefinition();
-                return true;
-            }
-
-            if (IsComparable(property.PropertyType))
-            {
-                var field = new ComparableFilterFieldDescriptor(
-                    Context, property);
-                definition = field.CreateDefinition();
-                return true;
-            }
-
-            if (DotNetTypeInfoFactory.IsListType(type))
-            {
-                if (!TypeInspector.Default.TryCreate(type, out Utilities.TypeInfo typeInfo))
-                {
-                    throw new ArgumentException(
-                        FilterResources.FilterArrayFieldDescriptor_InvalidType,
-                        nameof(property));
-                }
-
-                Type elementType = typeInfo.ClrType;
-                ArrayFilterFieldDescriptor field;
-
-                if (elementType == typeof(string)
-                    || elementType == typeof(bool)
-                    || typeof(IComparable).IsAssignableFrom(elementType))
-                {
-                    field = new ArrayFilterFieldDescriptor(
-                        Context,
-                        property,
-                        typeof(ISingleFilter<>).MakeGenericType(elementType));
-                }
-                else
-                {
-                    field = new ArrayFilterFieldDescriptor(Context, property, elementType);
-                }
-
-                definition = field.CreateDefinition();
-                return true;
-            }
-
-            if (type.IsClass)
-            {
-                var field = new ObjectFilterFieldDescriptor(
-                    Context, property, property.PropertyType);
-                definition = field.CreateDefinition();
-                return true;
-            }
+            IEnumerator<TryCreateImplicitFilter> enumerator
+                = _convention.GetImplicitFactories().GetEnumerator();
 
-            definition = null;
-            return false;
-        }
+            while (enumerator.MoveNext()
+                && !enumerator.Current(Context, type, property, _convention, out definition))
+            {/**/}
```

Comment:
what is that?

Reply:
Iterate through all implicit filter factories, takes the definition of the first one that returns true
Should i use for each and break?